src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/AspNetTelemetryEventSource.cs

```diff
@@ -37,7 +37,21 @@ public void ActivityStarted(Activity activity)
         {
             if (this.IsEnabled(EventLevel.Verbose, EventKeywords.All))
             {
-                this.ActivityStarted(activity?.Id);
+                // Accessing activity.Id here will cause the Id to be initialized
+                // before the sampler runs. This will result in Id not reflecting the
+                // correct sampling flags
+                // https://github.com/dotnet/runtime/issues/61857
+                string activityId = null;
+                if (activity == null)
+                {
+                    this.ActivityStarted(activityId);
+                }
+                else
+                {
+                    activityId = string.Concat("00-", activity.TraceId.ToHexString(), "-", activity.SpanId.ToHexString());
```

Comment:
Consider using one concat instead of two (might be good to to benchmark and see what's the actual perf/alloc difference).

Reply:
// * Summary *

BenchmarkDotNet=v0.13.1, OS=Windows 10.0.22000
Intel Core i7-8850H CPU 2.60GHz (Coffee Lake), 1 CPU, 12 logical and 6 physical cores
.NET SDK=6.0.100
  [Host]     : .NET 5.0.12 (5.0.1221.52207), X64 RyuJIT
  DefaultJob : .NET 5.0.12 (5.0.1221.52207), X64 RyuJIT


|    Method |     Mean |    Error |   StdDev |
|---------- |---------:|---------:|---------:|
| OneConcat | 66.48 ns | 1.381 ns | 2.455 ns |
| TwoConcat | 60.69 ns | 1.014 ns | 0.792 ns |

Reply:
looks like two concat works better.